src/util/triplea/MapXMLCreator/MapPropertiesPanel.java

```diff
@@ -0,0 +1,360 @@
+package util.triplea.MapXMLCreator;
+
+import java.awt.Dimension;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.awt.event.ActionEvent;
+import java.awt.event.FocusEvent;
+import java.awt.event.FocusListener;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.io.File;
+
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+
+import util.image.FileOpen;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class MapPropertiesPanel {
+	
+	public static void layout(final MapXMLCreator mapXMLCreator, final JPanel m_stepActionPanel) {
+
+		if (!MapXMLHelper.s_xmlStrings.containsKey("info_@name"))
+		{
+			for (final String newKey : new String[] { "info_@name", "info_@version" })
+			{
+				MapXMLHelper.s_xmlStrings.put(newKey, "");
+			}
+		}
+		
+		if (MapXMLHelper.s_resourceList.isEmpty())
+			MapXMLHelper.s_resourceList.add("");
+
+		final JTextField tMapName = new JTextField(MapXMLHelper.s_xmlStrings.get("info_@name"));

```

Comment:
Please minimize vertical distance between initialization and first usage.

Back 20 years ago this was good style:
```
int a, b, c, d;
a = 0;
b = a * 10;
d = a+b;
c = b / 20 -d;
```
The reason this is bad is because a reader has to remember 4 things while reading this code. Not to mention anything else they were trying to remember (and recall humans have a limit of about 7 or 9 things that they can remember at once). So add 3 more variables,, and a human is going to have to start looking up every variable to remember how it was used and defined. Contrast this with:


```
int a = 0;
int b = a * 10;
int d = a+b;
int c = b / 20 -d;
```
The latter is just easier..